gcloud/_apitools/transfer.py

```diff
@@ -774,6 +650,15 @@ def RefreshResumableUploadState(self):
         if self.strategy != RESUMABLE_UPLOAD:
             return
         self.EnsureInitialized()
+        # XXX Per RFC 2616/7231, a 'PUT' request is absolutely inappropriate
```

Comment:
Bait for you, actually:  I'm dumbfounded that the resumable upload protocol is so broken WRT standard HTTP semantics, and want you to be outraged, too.